src/windows.jl

```diff
@@ -1,46 +1,75 @@
 immutable WindowsPath <: AbstractPath
-    parts::Tuple
+    parts::Tuple{Vararg{String}}
     drive::String
+    root::String
 end
 
-WindowsPath() = WindowsPath(tuple(), "")
+WindowsPath() = WindowsPath(tuple(), "", "")
+
+WindowsPath(parts::Tuple) = WindowsPath(parts, "", "")
 
 function WindowsPath(str::AbstractString)
     if isempty(str)
-        return WindowsPath(tuple("."), "")
+        return WindowsPath(tuple("."), "", "")
     end
 
-    drive, path = splitdir(str)
-    tokenized = split(path, WIN_PATH_SEPARATOR)
-
-    if isempty(tokenized[1])
-        tokenized[1] = WIN_PATH_SEPARATOR
+    if startswith(str, "\\\\?\\")
+        error("The \\\\?\\ prefix is currently not supported.")
     end
 
-    return WindowsPath(tuple(map(String, tokenized)...), drive)
-end
+    str = replace(str, POSIX_PATH_SEPARATOR, WIN_PATH_SEPARATOR)
+
+    if startswith(str, "\\\\")
+        error("UNC paths are currently not supported.")
+    elseif startswith(str, "\\")
+        tokenized = split(str, WIN_PATH_SEPARATOR)
+
+        return WindowsPath(tuple(WIN_PATH_SEPARATOR, String.(tokenized[2:end])...), "", WIN_PATH_SEPARATOR)
+    elseif contains(str, ":")
+        l_drive, l_path = splitdrive(str)
 
+        tokenized = split(l_path, WIN_PATH_SEPARATOR)
 
-# The following should be implemented in the concrete types
-==(a::WindowsPath, b::WindowsPath) = parts(a) == parts(b) && drive(a) == drive(b)
+        l_root = isempty(tokenized[1]) ? WIN_PATH_SEPARATOR : ""
+
+        if isempty(tokenized[1])
+            tokenized = tokenized[2:end]
+        end
+
+        if !isempty(l_drive) || !isempty(l_root)
+            tokenized = tuple(string(l_drive, l_root), tokenized...)
+        end
+
+        return WindowsPath(tuple(String.(tokenized)...), l_drive, l_root)
+    else
+        tokenized = split(str, WIN_PATH_SEPARATOR)
+
+        return WindowsPath(tuple(String.(tokenized)...), "", "")
+    end
+end
+
+==(a::WindowsPath, b::WindowsPath) =
```

Comment:
I'd probably using 
```
function ==(...)
    ...
end
``` 
if we're going to split this across multiple lines.

Reply:
Done.